test/cpp/c10d/ProcessGroupNCCLErrorsTest.cpp

```diff
@@ -346,23 +347,64 @@ TEST_F(ProcessGroupNCCLErrorsTest, testNCCLErrorsNonBlocking) {
   // Communicators might be aborted here, further operations would fail.
 }
 
+// Function to read what we wrote to the local disk for validation.
+std::string readTraceInfoFromLocalDisk(int rank) {
```

Comment:
nit: just make 'filename' an arg here instead of doing the rank formatting inside?

Reply:
yeah I agree, and we can put this util function in `TestUtils.hpp` as well as `#include <fstream>`.

Reply:
I can put the name as an input arg. For @XilunWu's point, I do feel this is kind of specific to this test. And we an always revisit it when we see more use cases for this function.